modules/renter/hostdb/scan.go

```diff
@@ -206,12 +207,7 @@ func (hdb *HostDB) updateEntry(entry modules.HostDBEntry, netErr error) {
 
 	// Compress any old scans into the historic values.
 	for len(newEntry.ScanHistory) > minScans && time.Now().Sub(newEntry.ScanHistory[0].Timestamp) > maxHostDowntime {
-		timePassed := newEntry.ScanHistory[1].Timestamp.Sub(newEntry.ScanHistory[0].Timestamp)
-		if newEntry.ScanHistory[0].Success {
-			newEntry.HistoricUptime += timePassed
-		} else {
-			newEntry.HistoricDowntime += timePassed
-		}
+		decayUptimeOrDowntime(&newEntry, newEntry.ScanHistory[1], newEntry.ScanHistory[0])
```

Comment:
Upon further reflection (sorry for waffling around), the name should really be `updateUptime`, because you aren't just applying a decay you are also adding new information.